osrf_pycommon/process_utils/get_loop_impl.py

```diff
@@ -23,25 +23,34 @@ def get_loop_impl(asyncio):
     if getattr(_thread_local, 'loop_has_been_setup', False):
         return asyncio.get_event_loop()
```

Comment:
@clalancette  So the issue is actually here (related to this [comment](https://github.com/osrf/osrf_pycommon/pull/87/files#r1238640135)) let me explain a little better.

From my understanding, the method would only enter this if case when we set the member `loop_has_been_setup` to `True`, which we are doing at the end of this method after we set the event loop to `asyncio`. It would basically return the previously set loop instead of creating a new one. 
As we know the method `asyncio.get_event_loop()` have the deprecation warning on our current python version and should only be used on Python < 3.7. I went ahead and tried out using the appropriate method `asyncio.get_running_loop()` and when I was testing it out it throws a `RuntimeError` because there is no running event loop, that behavior is the one that seems weird to me according for what the if case should be.

Reply:
I noticed this error when running the test for the `osrf_pycommon`, `launch` and `test_launch_ros` packages. I'm aware that `launch` and `launch_ros` use the same method `asyncio.get_event_loop()` and that is generating warnings as expected, but, they also use this method from the `osrf_pycommon` package.

Reply:
> As we know the method `asyncio.get_event_loop()` have the deprecation warning on our current python version and should only be used on Python < 3.7. I went ahead and tried out using the appropriate method `asyncio.get_running_loop()` and when I was testing it out it throws a `RuntimeError` because there is no running event loop, that behavior is the one that seems weird to me according for what the if case should be.

Oh, I see.  So, I think the problem with doing that is that we are *not* in a coroutine nor in a callback, so it won't work.  See the documentation at https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_running_loop .  So I think we need to leave this as a `get_event_loop` call, which should *not* throw the warning because by the time we got here, we know that we already have an event loop setup.  See the documentation at https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop .

All of that is to say that I think the code as-is here is the right way to go.  But I could use another look from @cottsay or @sloretz , since this is the first time I'm dealing with this stuff.

Reply:
> So I think we need to leave this as a get_event_loop call, which should not throw the warning because by the time we got here, we know that we already have an event loop setup. See the documentation at https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop .

I agree with you that there shouldn't be any warning at this point of the code because the event loop have been setup and there should exist an event loop to return, however, we are still getting the warnings. For example look at the next picture (I'm using the code from this PR):
![image](https://github.com/osrf/osrf_pycommon/assets/51831786/320cf114-2c25-41db-9fd7-ebf8489e4d63)

We could implement the solution @cottsay initially proposed, but, on that part of the code in order to avoid the warning. However, the thing is that the reason why the warning appears could be a bug because we are not returning the event loop that was setup before right?

Reply:
I see.

After playing around with this for a bit, and reading the documentation, I'm now thinking that we should just take Scott's original patch.

The important point is the note in https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop, that I'll copy verbatim here:

```
Note

In Python versions 3.10.0–3.10.8 and 3.11.0 this function (and other functions which use it implicitly) emitted a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) if there was no running event loop, even if the current loop was set on the policy. In Python versions 3.10.9, 3.11.1 and 3.12 they emit a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) if there is no running event loop and no current loop is set. In some future Python release this will become an error.
```

And this is absolutely true; since Ubuntu 22.04 has 3.10.6, we are getting the warning with the existing code.  But if I run the existing code on Fedora 38, which has Python 3.11.3, no warning shows up.  That leads me to believe that the warning was actually too strict until Python 3.10.8, and Ubuntu 22.04 just happens to be in the middle of it.

Thus, I think we should go back to Scott's original patch and just suppress the warning.  It will be automatically suppressed in future Python versions, so I think this is all OK.  Thoughts?

Reply:
Oh I see what you mean. Agree about using Scott's initial patch. Should I create a new PR doing that? (Taking into account that the previous PR did the modification on the code and also merges the latest changes from rolling into this branch).